rudder-core/src/main/scala/com/normation/rudder/services/policies/write/AgentSpecificLogic.scala

```diff
@@ -120,9 +120,11 @@ class AgentRegister {
    * Find the first agent matching the required agentType/osDetail and apply f on it.
    * If none is found, return a failure.
    */
-  def findMap[T](agentType: AgentType, osDetails: OsDetails)(f: AgentSpecificGeneration => Box[T]) = {
-    pipeline.find(handler => handler.handle(agentType, osDetails)) match {
-      case None    => Failure(s"We were unable to find how to create directive sequences for Agent type ${agentType.toString()} on '${osDetails.fullName}'. " +
+  def findMap[T](agentNodeProps: AgentNodeProperties)(f: AgentSpecificGeneration => Box[T]) = {
+    pipeline.find(handler => handler.handle(agentNodeProps)) match {
+      case None    =>
+        val relayMsg = if(agentNodeProps.isPolicyServer) { "(scale-out relay server)"} else ""
+        Failure(s"We were unable to find how to create directive sequences for Agent type ${agentNodeProps.agentType.toString()} on '${agentNodeProps.osDetails.fullName}' ${relayMsg}. " +
```

Comment:
Maybe we could have a different message for policy server, clearer about the impact and what could be the cause (missing plugin ? ). I propose to change messages this way:

* non policy server:
We could not generate policies for node '${agentNodeProps.osDetails.fullName}' based on '${agentNodeProps.agentType.toString()}' agent. Maybe you are missing dedicated plugin ?

* policy server:
We could not generate policies for server '${agentNodeProps.osDetails.fullName}', therefore making updates for nodes behind it unavailable. Maybe you are missing "scale out" plugin?